k2/csrc/top_sort_test.cu

```diff
@@ -106,11 +106,90 @@ TEST(TopSort, VectorOfFsas) {
 }
 
 TEST(TopSort, RandomSingleFsa) {
-  // TODO(fangjun)
+  for (auto &context : {GetCpuContext(), GetCudaContext()}) {
+    Fsa fsa = GetRandFsa();
+    fsa = fsa.To(context);
+
+    int32_t gt = kFsaPropertiesTopSorted | kFsaPropertiesTopSortedAndAcyclic;
+    int32_t p = GetFsaBasicProperties(fsa);
+    EXPECT_NE(p & gt, gt);
+
+    Fsa sorted;
+    Array1<int32_t> arc_map;
+    TopSort(fsa, &sorted, &arc_map);
+
+    p = GetFsaBasicProperties(sorted);
+    EXPECT_EQ(p & gt, gt);
+
+    Array1<Arc> arcs = sorted.values.To(GetCpuContext());
+    arc_map = arc_map.To(GetCpuContext());
+    int32_t num_arcs = arcs.Dim();
+    for (int32_t i = 0; i != num_arcs; ++i) {
+      EXPECT_EQ(arcs[i].score, arc_map[i]);
+    }
+  }
 }
 
 TEST(TopSort, RandomVectorOfFsas) {
-  // TODO(fangjun)
+  int num_fsas = 4;
+  ContextPtr cpu = GetCpuContext();
+  for (auto &context : {GetCpuContext(), GetCudaContext()}) {
+    std::vector<Fsa> fsas(num_fsas);
+    for (int32_t i = 0; i != num_fsas; ++i) {
+      fsas[i] = GetRandFsa();
```

Comment:
What I'm thinking is we may have a better way to generate random FsaVec in k2: we can just call `RandomRaggedShape(true, 2,2, 0, max_arcs)` to generate meta info (i.e. states and num-arcs in each states) of each FSA, and then generate arcs according to those meta info (plus with configure parameters,e.g. sorted-or-not, empty-weights, etc.). Finally We can just call `Stack` to stack those FSAs into a FsaVec.
` RandomRaggedShape` can generate reasonable random shapes (including empty shapes), which is what we really need in a random test. BTW, usually I would run the random tests in a for-loop (e.g. `i in range(100)`) locally to have a reliable result (and may change to `i in range(2)`) when committing code.